virtcontainers/clh.go

```diff
@@ -165,98 +136,77 @@ func (clh *cloudHypervisor) createSandbox(ctx context.Context, id string, networ
+		// cast to int32, this is a openAPI limitation that not support unsigned values
+		CpuCount: int32(clh.config.NumVCPUs),
+	}
 
 	// Add the kernel path
 	kernelPath, err := clh.config.KernelAssetPath()
 	if err != nil {
 		return err
 	}
-	clh.cliBuilder.SetKernel(&CLIKernel{
-		path: kernelPath,
-	})
+	clh.vmconfig.Kernel = chclient.KernelConfig{
+		Path: kernelPath,
+	}
 
 	// First take the default parameters defined by this driver
-	clh.cliBuilder.AddKernelParameters(clhKernelParams)
+	kParams := clhKernelParams
 
 	// Followed by extra debug parameters if debug enabled in configuration file
 	if clh.config.Debug {
-		clh.cliBuilder.AddKernelParameters(clhDebugKernelParams)
+		kParams = append(kParams, clhDebugKernelParams...)
 	}
 
 	// Followed by extra debug parameters defined in the configuration file
-	clh.cliBuilder.AddKernelParameters(clh.config.KernelParams)
+
+	kParams = append(kParams, clh.config.KernelParams...)
+
+	clh.vmconfig.Cmdline.Args = kernelParamsToString(kParams)
 
 	// set random device generator to hypervisor
-	clh.cliBuilder.SetRng(&CLIRng{
-		src:   clh.config.EntropySource,
-		iommu: false,
-	})
+	clh.vmconfig.Rng = chclient.RngConfig{
+		Src: clh.config.EntropySource,
+	}
 
 	// set the initial root/boot disk of hypervisor
 	imagePath, err := clh.config.ImageAssetPath()
 	if err != nil {
 		return err
 	}
 
-	if imagePath != "" {
-		clh.cliBuilder.SetDisk(&CLIDisk{
-			path:  imagePath,
-			iommu: false,
-		})
+	if imagePath == "" {
+		return errors.New("image path is empty")
 	}
 
-	// set the virtio-fs to the hypervisor
-	vfsdSockPath, err := clh.virtioFsSocketPath(clh.id)
-	if err != nil {
-		return err
-	}
-	if clh.config.VirtioFSCache == virtioFsCacheAlways {
-		clh.cliBuilder.SetFs(&CLIFs{
-			tag:        "kataShared",
-			socketPath: vfsdSockPath,
-			queues:     1,
-			queueSize:  512,
-			dax:        true,
-		})
-	} else {
-		clh.cliBuilder.SetFs(&CLIFs{
-			tag:        "kataShared",
-			socketPath: vfsdSockPath,
-			queues:     1,
-			queueSize:  512,
-		})
+	disk := chclient.DiskConfig{
+		Path: imagePath,
 	}
+	clh.vmconfig.Disks = append(clh.vmconfig.Disks, disk)
 
 	// set the serial console to the cloud hypervisor
 	if clh.config.Debug {
 		serialPath, err := clh.serialPath(clh.id)
 		if err != nil {
 			return err
 		}
-		clh.cliBuilder.SetSerial(&CLISerialConsole{
-			consoleType: cctFILE,
-			filePath:    serialPath,
-		})
-		logFilePath, err := clh.logFilePath(clh.id)
-		if err != nil {
-			return err
+		clh.vmconfig.Serial = chclient.ConsoleConfig{
+			Mode: cctFILE,
+			File: serialPath,
+		}
+
+	} else {
+		clh.vmconfig.Serial = chclient.ConsoleConfig{
+			Mode: cctOFF,
 		}
-		clh.cliBuilder.SetLogFile(&CLILogFile{
-			path: logFilePath,
-		})
 	}
 
-	clh.cliBuilder.SetConsole(&CLIConsole{
-		consoleType: cctOFF,
-	})
+	clh.vmconfig.Console = chclient.ConsoleConfig{
+		Mode: cctOFF,
+	}
 
 	// Move the API endpoint socket location for the
 	// by default enabled api endpoint
```

Comment:
this isn't clear to me -- can you clarify this comment?

Reply:
I believe the comment means: "Overwrite the default value of HTTP API socket path for cloud hypervisor"

Reply:
Let's update with that clarity?